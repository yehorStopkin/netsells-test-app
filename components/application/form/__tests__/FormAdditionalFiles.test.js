const { shallowMount } = require('@vue/test-utils');
const FormAdditionalFiles = require('@/components/application/form/FormAdditionalFiles').default;

describe('FormAdditionalFiles', () => {
    getValue = (cv = null, coverLetter = null) => {
        return {
            cv,
            coverLetter,
        };
    };

    const uploadedFile = {
        name: 'Test file name.pdf',
    };

    test('component default', () => {
        const wrapper = shallowMount(FormAdditionalFiles, { propsData: { value: getValue(), step: 2 } });
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();
    });

    test('component uploaded cv', () => {
        const wrapper = shallowMount(FormAdditionalFiles, { propsData: { value: getValue(uploadedFile), step: 2 } });
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();
    });

    test('component uploaded cover letter', () => {
        const wrapper = shallowMount(FormAdditionalFiles, { propsData: { value: getValue(null, uploadedFile), step: 2 } });
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();
    });
});
