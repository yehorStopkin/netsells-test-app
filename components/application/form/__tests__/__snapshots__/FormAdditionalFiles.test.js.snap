// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAdditionalFiles component default 1`] = `
<div
  class="form-additional-files"
>
  <form-header-stub
    class="form-additional-files__header"
    title="Files Upload"
  />
   
  <div
    class="form-additional-files__field"
  >
    <div
      action=""
      drag=""
      on-change="function () { [native code] }"
    >
      <div
        class="form-additional-files__field_empty"
      >
        <div
          class="form-additional-files__field-icon required"
        />
         
        <div
          class="form-additional-files__field-title"
        >
          
                    Upload Your CV
                
        </div>
         
        <div
          class="form-additional-files__field-subtitle"
        >
          <span>
            Drag and drop or Upload your CV file
          </span>
           
          <span
            class="form-additional-files__field-upload-link"
          >
            
                        here
                    
          </span>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="form-additional-files__field"
  >
    <div
      action=""
      drag=""
      on-change="function () { [native code] }"
    >
      <div
        class="form-additional-files__field_empty"
      >
        <div
          class="form-additional-files__field-icon"
        />
         
        <div
          class="form-additional-files__field-title"
        >
          
                    Upload Your Cover Letter
                
        </div>
         
        <div
          class="form-additional-files__field-subtitle"
        >
          <span>
            Drag and frop or Upload your Cover Letter file
          </span>
           
          <span
            class="form-additional-files__field-upload-link"
          >
            
                        here
                    
          </span>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="form-additional-files__actions"
  >
    <div
      text="NEXT"
      width="115"
    />
  </div>
</div>
`;

exports[`FormAdditionalFiles component uploaded cover letter 1`] = `
<div
  class="form-additional-files"
>
  <form-header-stub
    class="form-additional-files__header"
    title="Files Upload"
  />
   
  <div
    class="form-additional-files__field"
  >
    <div
      action=""
      drag=""
      on-change="function () { [native code] }"
    >
      <div
        class="form-additional-files__field_empty"
      >
        <div
          class="form-additional-files__field-icon required"
        />
         
        <div
          class="form-additional-files__field-title"
        >
          
                    Upload Your CV
                
        </div>
         
        <div
          class="form-additional-files__field-subtitle"
        >
          <span>
            Drag and drop or Upload your CV file
          </span>
           
          <span
            class="form-additional-files__field-upload-link"
          >
            
                        here
                    
          </span>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="form-additional-files__field"
  >
    <div
      action=""
      drag=""
      on-change="function () { [native code] }"
    >
      <div
        class="form-additional-files__field_uploaded"
      >
        <div
          class="form-additional-files__field-title-wrapper"
        >
          <div
            class="form-additional-files__field-icon"
          />
           
          <div
            class="form-additional-files__field-title"
            title="Test file name.pdf"
          >
            
                        Test file name.pdf
                    
          </div>
        </div>
         
        <div
          class="form-additional-files__field-subtitle"
        >
          <span>
            Reupload a new file
          </span>
           
          <span
            class="form-additional-files__field-upload-link"
          >
            
                        here
                    
          </span>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="form-additional-files__actions"
  >
    <div
      text="NEXT"
      width="115"
    />
  </div>
</div>
`;

exports[`FormAdditionalFiles component uploaded cv 1`] = `
<div
  class="form-additional-files"
>
  <form-header-stub
    class="form-additional-files__header"
    title="Files Upload"
  />
   
  <div
    class="form-additional-files__field"
  >
    <div
      action=""
      drag=""
      on-change="function () { [native code] }"
    >
      <div
        class="form-additional-files__field_uploaded"
      >
        <div
          class="form-additional-files__field-title-wrapper"
        >
          <div
            class="form-additional-files__field-icon"
          />
           
          <div
            class="form-additional-files__field-title"
            title="Test file name.pdf"
          >
            
                        Test file name.pdf
                    
          </div>
        </div>
         
        <div
          class="form-additional-files__field-subtitle"
        >
          <span>
            Reupload a new file
          </span>
           
          <span
            class="form-additional-files__field-upload-link"
          >
            
                        here
                    
          </span>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="form-additional-files__field"
  >
    <div
      action=""
      drag=""
      on-change="function () { [native code] }"
    >
      <div
        class="form-additional-files__field_empty"
      >
        <div
          class="form-additional-files__field-icon"
        />
         
        <div
          class="form-additional-files__field-title"
        >
          
                    Upload Your Cover Letter
                
        </div>
         
        <div
          class="form-additional-files__field-subtitle"
        >
          <span>
            Drag and frop or Upload your Cover Letter file
          </span>
           
          <span
            class="form-additional-files__field-upload-link"
          >
            
                        here
                    
          </span>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="form-additional-files__actions"
  >
    <div
      text="NEXT"
      width="115"
    />
  </div>
</div>
`;
